### PR TITLE
[DOCS] EQL: Document `indexOf` function

### PR DESCRIPTION
**Pull request by jrodewig**
Documents the `indexOf` EQL function.

Relates to #54543 and #53856

### CODE REVIEW

File: docs/reference/eql/functions.asciidoc [hunk 1]
```diff
@@ -207,6 +208,110 @@ field datatypes:
...
+--
+
+`<start_pos>`::
++
+--
+(Optional, integer or `null`)
+Starting position for matching. The function will not return positions before
+this one. Defaults to `0`.
+
+Positions are zero-indexed. Negative offsets are supported.
```
Comment (astefan) [inline docs/reference/eql/functions.asciidoc:298]: Could you add an example with a negative offset?

**Review by astefan — APPROVED** [review]
LGTM. Left just one minor comment.

File: docs/reference/eql/functions.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/eql/functions.asciidoc:300]: Sure thing. Added some examples with db8e027.

**Comment by jrodewig** [comment]
### Backport commits
master 881b214 
7.x 4f2ab96f389

**Pull request merged**